include/cutlass/gpu_generics.h

```diff
@@ -248,10 +248,11 @@ unsigned int byte_perm(unsigned int x, unsigned int y, unsigned int s) {
 
 // shfl
 
+template<typename T>
```

Comment:
Perhaps we need a `static_assert` in each to confirm T is a [supported type](https://docs.nvidia.com/cuda/cuda-c-programming-guide/#warp-shuffle-synopsis)?

Reply:
I think we are better off without such a check. That way we can instantiate the template with unsupported type that can be implicitly cast into a supported one, which is the same behaviour as calling the shuffle intrinsic directly.